src/k8s/pkg/utils/file_test.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestPath(t *testing.T) {
+	t.Setenv("SNAP", "snapenv")
+	t.Setenv("SNAP_DATA", "/data/snapenv")
+	t.Setenv("SNAP_COMMON", "common/snapenv")
+	for _, tc := range []struct {
+		name          string
+		input_path    []string
+		expected_path string
+	}{
+		{
+			name:          "nil",
+			expected_path: "snapenv",
+		},
+		{
+			name:          "abc",
+			input_path:    []string{"abc"},
+			expected_path: "snapenv/abc",
+		},
+		{
+			name:          "abc/def",
+			input_path:    []string{"abc", "def"},
+			expected_path: "snapenv/abc/def",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			simple := tc.expected_path
+			data := fmt.Sprintf("/data/%s", tc.expected_path)
+			common := fmt.Sprintf("common/%s", tc.expected_path)
+			if parsed := Path(tc.input_path...); !reflect.DeepEqual(parsed, simple) {
+				t.Fatalf("expected path to be %v but it was %v instead", simple, parsed)
+			}
+			if parsed := DataPath(tc.input_path...); !reflect.DeepEqual(parsed, data) {
+				t.Fatalf("expected data path to be %v but it was %v instead", data, parsed)
+			}
+			if parsed := CommonPath(tc.input_path...); !reflect.DeepEqual(parsed, common) {
+				t.Fatalf("expected common path to be %v but it was %v instead", common, parsed)
+			}
```

Comment:
Yup, I like this test. Note that we found it gets very tedious very quickly to write tests like this, can we bring in gomega instead?

See for example https://github.com/canonical/microk8s-cluster-agent/blob/main/pkg/k8sinit/launcher_test.go

The test would then like this:

```
g := NewWithT(t)
g.Expect(Path(tc.input_path...)).To(Equal(tc.expected_path))
g.Expect(DataPath(tc.input_path...)).To(Equal(fmt.Sprintf("/data/%s", tc.expected_path))
g.Expect(CommonPath(tc.input_path...)).To(Equal(fmt.Sprintf("common/%s", tc.expected_path))
```

This also gives us much nicer output and failure messages when something breaks.